### DISCUSSION TITLE
Does duckdb support write & ready with multiple threads?

### DISCUSSION THREAD

**Discussion by portsip** [duckdb/duckdb#13749] [General]
Hi,

I have read this guide: https://duckdb.org/docs/connect/concurrency.html but I still want to ask help.

We would like to use DuckDB to store our PBX CDR (Call Detail Records) data. The usage would be as follows:

- Process A continuously produces the data and writes it into DuckDB.
- Another process reads the data from DuckDB simultaneously as Process A writes and sends data to the browser to display as charts.

Does DuckDB support this workflow?

Thanks.

**Comment by imxade** [comment]
https://duckdb.org/docs/connect/concurrency#writing-to-duckdb-from-multiple-processes
Not straight forward tho, otherwise this would've been a silver bullet.
